### PR TITLE
KOA-5825 Migrate dimension and duration token generation to Kotlin

### PR DESCRIPTION
**Pull request by marianeum**
+ [x] Check this if you have read and followed the [contributing guidelines](https://github.com/Skyscanner/backpack-android/blob/main/CONTRIBUTING.md)

Remember to include the following changes:
+ [x] Component `README.md`
+ [x] Tests
+ [x] Docs (either update [backpack-docs](https://github.com/Skyscanner/backpack-docs) now, or create a follow up ticket)

If you are curious about how we review, please read through the [code review guidelines](https://github.com/Skyscanner/backpack/blob/main/CODE_REVIEW_GUIDELINES.md)

### CODE REVIEW

File: Backpack/src/main/res/values/backpack.animation.xml [hunk 1]
```diff
@@ -17,11 +17,7 @@
...
 -->
 <resources>
-
-	<integer name="bpkAnimationDurationXs">50</integer>
-
-	<integer name="bpkAnimationDurationSm">200</integer>
-
-	<integer name="bpkAnimationDurationBase">400</integer>
-
+  <integer name="bpkAnimationDurationXs">50</integer>
```
Comment (marianeum) [inline Backpack/src/main/res/values/backpack.animation.xml:20]: just fixed the formatting which was always off

File: Backpack/src/main/res/values/backpack.dimensions.spacing.xml [hunk 2]
```diff
@@ -17,17 +17,11 @@
...
-	<dimen name="bpkSpacingXxl">40dp</dimen>
-
- </resources>
+  <dimen name="bpkSpacingSm">4dp</dimen>
+  <dimen name="bpkSpacingMd">8dp</dimen>
+  <dimen name="bpkSpacingBase">16dp</dimen>
+  <dimen name="bpkSpacingLg">24dp</dimen>
+  <dimen name="bpkSpacingXl">32dp</dimen>
+  <dimen name="bpkSpacingXxl">40dp</dimen>
+  <dimen name="bpkSpacingNone">0dp</dimen>
```
Comment (marianeum) [inline Backpack/src/main/res/values/backpack.dimensions.spacing.xml:26]: none was filtered out before for some reason, but that would require more reworking of the current system - I don't see harm in having it

File: backpack-compose/src/main/kotlin/net/skyscanner/backpack/compose/tokens/BpkBorderSize.kt [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+// Auto-generated: do not edit
+@file:Suppress("RedundantVisibilityModifier","unused")
+
+package net.skyscanner.backpack.compose.tokens
+
+import androidx.compose.ui.unit.Dp
+import androidx.compose.ui.unit.dp
+
+public object BpkBorderSize {
```
Comment (marianeum) [inline backpack-compose/src/main/kotlin/net/skyscanner/backpack/compose/tokens/BpkBorderSize.kt:27]: this was actually missing 😬

File: buildSrc/src/main/resources/resource_file_template.txt [hunk 4]
```diff
@@ -17,7 +17,5 @@
   limitations under the License.
 -->
 <resources>
-{% for animation in data %}
-	<integer name="{{animation.name}}">{{animation.value}}</integer>
-{% endfor %}
+{{content}}
```
Comment (marianeum) [inline buildSrc/src/main/resources/resource_file_template.txt:20]: the git diff is a bit confusing, but basically this is the construct of any resource files. the ``{{content}}` bit is replaced in code

File: gulpfile.js [hunk 5]
```diff
@@ -363,36 +286,12 @@ gulp.task('template:icons', () =>
     .pipe(gulp.dest(PATHS.drawableRes)),
 );
 
-gulp.task('template:animation', () => {
```
Comment (frugoman) [inline gulpfile.js:366]: 🔥

**Pull request merged**